swap_meet/vendor.py

```diff
@@ -1,2 +1,82 @@
+from .item import Item
+
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+        self.item = Item()
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+
+    def swap_items(self, vendor, my_item, their_item):
+        if their_item in vendor.inventory and my_item in self.inventory:
+            self.inventory.append(their_item)
+            vendor.inventory.append(my_item)
+            vendor.inventory.remove(their_item)
+            self.inventory.remove(my_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, vendor):
+        if len(self.inventory) != 0 and len(vendor.inventory) != 0:
+            return self.swap_items(vendor, self.inventory[0], vendor.inventory[0])
+        else:
+            return False
+
+    def get_best_by_category(self, category):
+        items_by_category = self.get_by_category(category)
+        if items_by_category == None:
+            return None
+
+        # initatied variables and looped over dictionary instead of using max. wanted to consider 
+        # the possibility in the future that there may be more than one max and I wanted the data 
+        # to be organized in a dictionary because I think it makes more sense. Please
+        # share your thoughts on this function aprticularly if you have some, thank you!
+
+        top_item_condition = 0
+        top_item = None
+        best_by_category = {}
+        
+        for item in items_by_category:
+            best_by_category[item] = item.condition
+        for item, item_condition in best_by_category.items():
+            if item_condition > top_item_condition:
+                top_item_condition = item_condition
+                top_item = item
+        return top_item
```

Comment:
I would say since the README states that the method will return only one regardless of "ties," we can remove `best_by_category.` 

If there was a possibility for ties, I think you would still only want to add the "best" into a list or tuple instead of turning the entire instance of the Item into a key, because 1) the Item's condition is already inside the instance (no need to then restate the condition as a second value and 2) you must still iterate through the original list `items_by_category`, so you still touching each item at least once.

So I'd say it isn't saving time or space complexity by doing so.

```suggestion
        top_item = items_by_category[0]
        
        for item in items_by_category:
            if item.condition > top_item.condition:
                top_item = item
        return top_item
```